src/xmipp/applications/programs/mpi_nma_alignment_vol/mpi_nma_alignment_vol_main.cpp

```diff
@@ -0,0 +1,140 @@
+/***************************************************************************
+ *
+ * Authors:  Slavica Jonic slavica.jonic@impmc.jussieu.fr
+ *           Carlos Oscar Sanchez Sorzano coss.eps@ceu.es
+ *
+ * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+ * Lab. de Bioingenieria, Univ. San Pablo CEU
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.uam.es'
+ ***************************************************************************/
+//#include <mpi.h>
+#include <parallel/xmipp_mpi.h>
+#include <reconstruction/nma_alignment_vol.h>
+
+
+/** Class to perfom the NMA Alignment with  MPI parallelization */
+class MpiProgNMAVol: public ProgNmaAlignmentVol
+{
+private:
+    MpiNode *node;
+    MpiTaskDistributor *distributor;
+    std::vector<size_t> imgsId;
+    MpiFileMutex *fileMutex;
+
+public:
+    /** Empty Constructor */
+    MpiProgNMAVol()
+    {
+    	node=NULL;
+    	distributor=NULL;
+    	fileMutex=NULL;
+    }
+
+    /** Destructor */
+    ~MpiProgNMAVol()
+    {
+        delete node;
+        delete fileMutex;
+        delete distributor;
+    }
+
+    /** Redefine read to initialize MPI environment */
+    void read(int argc, char **argv)
+    {
+        node = new MpiNode(argc, argv);
+        if (!node->isMaster())
+        	verbose=0;
+        fileMutex = new MpiFileMutex(node);
+        ProgNmaAlignmentVol::read(argc, argv);
+    }
+    /** main body */
+    void createWorkFiles()
+    {
+        //Master node should prepare some stuff before start working
+        MetaData &mdIn = *getInputMd(); //get a reference to input metadata
+
+        if (node->isMaster())
+        {
+            ProgNmaAlignmentVol::createWorkFiles();
+            mdIn.write(fnOutDir + "/nmaTodo.xmd");
+        }
+        node->barrierWait();//Sync all before start working
+        mdIn.read(fnOutDir + "/nmaTodo.xmd");
+        mdIn.findObjects(imgsId);//get objects ids
+        rangen = node->rank;
```

Comment:
What is rangen? From the description it's a random generator seed, but it's never used that way. Please fix it.

Reply:
The name originates to nma_alignment.cpp and nma_alignment_vol.cpp and the authors had a justification for the name. If at some point this name changes over there, it would follow here. For the moment, let's keep it as it is.

Reply:
in `nma_alignment.cpp` it is never used as a random generator seed. Please rename it approprietly